sui_programmability/framework/sources/Governance/ValidatorSet.move

```diff
@@ -265,9 +329,17 @@ module Sui::ValidatorSet {
         }
     }
 
-    #[test_only]
-    public fun get_total_stake(self: &ValidatorSet): u64 {
-        self.total_stake
+    fun distribute_reward(validators: &mut vector<Validator>, total_stake: u64, reward: &mut Coin<SUI>, ctx: &mut TxContext) {
+        let total_reward_amount = Coin::value(reward);
+        let length = Vector::length(validators);
+        let i = 0;
+        while (i < length) {
+            let validator = Vector::borrow_mut(validators, i);
+            let reward_amount = Validator::get_stake_amount(validator) * total_reward_amount / total_stake;
```

Comment:
I don't know enough about the tokenomics math here, but it feels like there should be some assertion about the `Coin::value(reward)` at the end. 
Mostly this division math makes me uncomfortable (in that there could be some unexpected remainder at the end) 

If no need for an assertion, maybe just a comment explaining this?